### ISSUE TITLE
The method '+' was called on null

### ISSUE THREAD

**Issue by Holofox** [kushalmahapatro/infospect#3]
Hello! Thanks for the package! There is a problem with form data requests.
```
NoSuchMethodError: The method '+' was called on null.\nReceiver: null\nTried calling: +("Form data")]
```
I'm guessing the problem is on this line:
https://github.com/kushalmahapatro/infospect/blob/157e00476c67998c93d2c0355abcd29add6500f0/lib/features/network/interceptors/infospect_dio_interceptor.dart#L51

I also think it’s worth wrapping all the handlers in the interceptor in `try/catch`, so that in case of an error this does not affect the sending of the request. The error can be made non-blocking via `handler.next()` so that other interceptors can continue working.

https://github.com/kushalmahapatro/infospect/blob/157e00476c67998c93d2c0355abcd29add6500f0/lib/features/network/interceptors/infospect_dio_interceptor.dart#L39

https://github.com/kushalmahapatro/infospect/blob/157e00476c67998c93d2c0355abcd29add6500f0/lib/features/network/interceptors/infospect_dio_interceptor.dart#L111

https://github.com/kushalmahapatro/infospect/blob/157e00476c67998c93d2c0355abcd29add6500f0/lib/features/network/interceptors/infospect_dio_interceptor.dart#L145

**Comment by kushalmahapatro** [comment]
Hey @Holofox , Thanks for pointing this out, 
the body param is not yet initialized and +("Form data") was being appended to it. 

Will fix this and let you know and I agree to your suggestion to handle the interceptors with try/catch, so that it wont hamper the Call, if the interception failed. 

``` dart
NoSuchMethodError: The method '+' was called on null.\nReceiver: null\nTried calling: +("Form data")]
```

**Comment by kushalmahapatro** [comment]
@Holofox , The fix has been added in 0.1.2.
